src/NBitcoin/Rules/IConsensusRule.cs

```diff
@@ -1,6 +1,22 @@
 ﻿namespace NBitcoin.Rules
 {
-    public interface IConsensusRule
+    public interface IConsensusRuleBase
+    {
+    }
+
+    public interface IHeaderValidationConsensusRule : IConsensusRuleBase
```

Comment:
I dont think we need interfaces for each type of collection, just make the distinction by name.

Reply:
I actually think this is a good one